idea/data-storage/src/rabbitmq/rabbitmq.service.ts

```diff
@@ -0,0 +1,160 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Channel, Connection, connect, Replies } from 'amqplib';
+import {
+  AddMetaParams,
+  API_METHODS, FindMessageParams,
+  FindProgramParams, GetAllCodeParams,
+  GetAllProgramsParams, GetCodeParams, GetMessagesParams, GetMetaParams,
+  RabbitMQExchanges,
+  RabbitMQueues,
+} from '@gear-js/common';
+
+import { ProgramService } from '../program/program.service';
+import { MessageService } from '../message/message.service';
+import { MetadataService } from '../metadata/metadata.service';
+import { CodeService } from '../code/code.service';
+import { BlockService } from '../block/block.service';
+import { RabbitmqMessageParams } from './types/rabbitmq-params';
+import { FormResponse } from '../decorator/form-response.decorator';
+
+@Injectable()
+export class RabbitmqService {
+  private logger: Logger = new Logger(RabbitmqService.name);
+  private mainChannel: Channel;
+  private topicChannel: Channel;
+  private connection: Connection;
+
+  constructor(
+    private configService: ConfigService,
+    private messageService: MessageService,
+    private metaService: MetadataService,
+    private codeService: CodeService,
+    private programService: ProgramService,
+    private blockService: BlockService,
+  ) {}
+
+  public async connect(): Promise<void> {
+    this.connection = await connect(this.configService.get<string>('rabbitmq.url'));
+    console.log('📍 RabbitMQ connected successfully');
+  }
+
+  public async initRMQ(genesis: string): Promise<void> {
+    try {
+      this.mainChannel = await this.connection.createChannel();
+      this.topicChannel = await this.connection.createChannel();
+
+      const directExchange = RabbitMQExchanges.DIRECT_EX;
+      const topicExchange = RabbitMQExchanges.TOPIC_EX;
+      const directExchangeType = 'direct';
+      const routingKey = `ds.${genesis}`;
+
+      //send genesis to api-gateway
+      const messageBuff = JSON.stringify({ service: 'ds', action: 'add', genesis });
+      this.mainChannel.publish(directExchange, RabbitMQueues.GENESISES, Buffer.from(messageBuff));
+
+      await this.mainChannel.assertExchange(directExchange, directExchangeType);
+      await this.topicChannel.assertExchange(topicExchange, 'topic');
+
+      const assertQueue = await this.mainChannel.assertQueue(`ds.AQ.${genesis}`, {
```

Comment:
What does `AQ` mean?

Reply:
`Assert Queue` . But it's not good idea, I know that. `ds.someGenesis` it's already declared name. Maybe better to rename like `dsq.someGenesis` and `dstq.someGenesis` ?

Reply:
But where do you use these queues? I mean, if all messages are sending to ds.genesis queue, what is the point of using another one queue?